tools/bazel.rc

```diff
@@ -1,6 +1,7 @@
 # Envoy specific Bazel build/test options.
 
 build --workspace_status_command=bazel/get_workspace_status
+build --experimental_cc_skylark_api_enabled_packages=@rules_foreign_cc//tools/build_defs,tools/build_defs
```

Comment:
This is the main blocker to use merging this PR; we don't want folks to have to be setting this when importing Envoy to their own projects. CC @irengrig.

Reply:
@irengrig this aged out of our review process, because we haven't figured out these last few steps and weren't able to verify Windows support (@sesmith177). Do you think there is any way to resolve these?

Reply:
@htuch we have used a version of bazel that has the fix for https://github.com/bazelbuild/bazel/issues/6292 and verified that we can build Envoy with it.

We have not had a chance yet to take another look at this PR

Reply:
Hi @htuch, I asked the people from Bazel C++ team, they estimate the work for getting rid of the --experimental_cc_skylark_api_enabled_packages flag as at least two weeks.
The task is their priority, but it takes time.

Nice to hear about Windows build.

Reply:
Thanks @irengrig. Please ping me when this is done and we will try and switch this on in master (hopefully @sesmith177 and MSFT will have time to evaluate by then).

Reply:
it gets farther on windows with the newer bazel, but still fails to build:

```
ERROR: C:/vmfiles/envoy/bazel/foreign_cc/BUILD:25:1: no such package '@com_github_nghttp2_nghttp2//': Traceback (most recent call last):
        File "C:/_eb/external/bazel_tools/tools/build_defs/repo/http.bzl", line 55
                patch(ctx)
        File "C:/_eb/external/bazel_tools/tools/build_defs/repo/utils.bzl", line 85, in patch
                fail(("Error applying patch command %...)))
Error applying patch command find . -name '*.sh' -exec sed -i.orig '1s|#!/usr/bin/env sh\$|/bin/sh\$|' {} +:

---------- LTMAIN.SH
Access denied - .
File not found - -NAME
File not found - -EXEC
File not found - SED
File not found - -I.ORIG
File not found - {}
File not found - +
 and referenced by '//bazel/foreign_cc:nghttp2'
```

we haven't had too much time to debug this